src/Stratis.Bitcoin.IntegrationTests/MinerTests.cs

```diff
@@ -887,5 +887,85 @@ private void MinerCreateNewBlockValidity()
             //    SetMockTime(0);
             //    mempool.clear();
         }
+
+        [Fact]
+        public void MiningAndPropagatingPOW_MineBlockCheckNodeConsensusTipIsCorrect()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode miner = builder.CreateStratisPowNode(KnownNetworks.RegTest);
+                builder.StartAll();
+                miner.NotInIBD().SetDummyMinerSecret(new BitcoinSecret(new Key(), miner.FullNode.Network));
+                miner.GenerateStratisWithMiner(5);
+
+                Assert.Equal(5, miner.FullNode.ConsensusManager().Tip.Height);
+            }
+        }
+
+        [Fact]
+        public void MiningAndPropagatingPOW_MineBlockCheckPeerHasNewBlock()
+        {
+            using (NodeBuilder builder = NodeBuilder.Create(this))
+            {
+                CoreNode miner = builder.CreateStratisPowNode(KnownNetworks.RegTest);
+                CoreNode[] syncers = new CoreNode[]
+                {
+                    builder.CreateStratisPowNode(KnownNetworks.RegTest),
+                    builder.CreateStratisPowNode(KnownNetworks.RegTest),
+                    builder.CreateStratisPowNode(KnownNetworks.RegTest)
+                };
+
+                builder.StartAll();
+
+                foreach (var syncer in syncers)
+                {
+                    miner.CreateRPCClient().AddNode(syncer.NotInIBD().Endpoint, true);
+                }
+
+                miner.NotInIBD().SetDummyMinerSecret(new BitcoinSecret(new Key(), miner.FullNode.Network));
+
+                Assert.True(syncers.All(x => x.FullNode.ConsensusManager().Tip.Height == 0));
+
+                miner.GenerateStratisWithMiner(3);
+
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(miner, syncers[0]));
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(miner, syncers[1]));
+                TestHelper.WaitLoop(() => TestHelper.AreNodesSynced(miner, syncers[2]));
+
+                Assert.True(syncers.All(x => x.FullNode.ConsensusManager().Tip.Height == 3));
+            }
+        }
+
+        [Fact]
+        public void MiningAndPropagatingPOW_NodeConsensusTipHasLessWorkThanNetworkTipResets()
```

Comment:
This test is not goo enough, we need to find a way to mine a block while consensus tip advances, its going to be a bit tricky perhaps @fassadlr can help in that.

We need to mine a block and not push it to our consensus code (the method CM.BlockMined) until a better block comes then the new block should be rejected.